src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -215,19 +215,78 @@ class DriftTable extends Component {
         return row;
     }
 
-    renderSortButton(sort) {
+    renderSortButton(sortType, sort) {
         let sortIcon;
 
         if (sort === 'asc') {
-            sortIcon = <AngleUpIcon className="pointer active-blue" onClick={ () => this.props.toggleFactSort('desc') }/>;
+            sortIcon = <ArrowUpIcon className="pointer active-blue" onClick={ () => this.toggleSort(sortType, sort) }/>;
         }
         else if (sort === 'desc') {
-            sortIcon = <AngleDownIcon className="pointer active-blue" onClick={ () => this.props.toggleFactSort('asc') }/>;
+            sortIcon = <ArrowDownIcon className="pointer active-blue" onClick={ () => this.toggleSort(sortType, sort) }/>;
+        }
+        else {
+            sortIcon = <ArrowsAltVIcon className="pointer" onClick={ () => this.toggleSort(sortType, sort) }/>;
         }
 
         return sortIcon;
     }
 
+    toggleSort(sortType, sort) {
+        if (sort === 'asc') {
+            if (sortType === 'fact') {
+                this.props.toggleFactSort('desc');
+            } else {
+                this.props.toggleStateSort('desc');
+            }
+        }
+        else if (sort === 'desc') {
+            if (sortType === 'fact') {
+                this.props.toggleFactSort('asc');
+            } else {
+                this.props.toggleStateSort('');
+            }
+        }
+        else {
+            this.props.toggleStateSort('asc');
+        }
+
+        this.props.toggleActiveSort(sortType);
+    }
+
+    renderHeaderRow(systems, loading) {
+        if (this.props.activeSort === 'fact' || this.props.stateSort === '') {
+            return (
+                <tr className="sticky-column-header">
+                    <th className="active-sort fact-header sticky-column fixed-column-1">
```

Comment:
As far as I can tell, the only difference between the if-else is the addition of this class on one of the two table headers. Can we drop the if-else and make inclusion of this class the part triggered by logic?